source/qb64.bas

```diff
@@ -1071,31 +1071,41 @@ IF C = 9 THEN 'run
 
     'execute program
 
-    IF iderunmode = 1 THEN
-        IF os$ = "WIN" THEN SHELL _DONTWAIT QuotedFilename$(CHR$(34) + lastBinaryGenerated$ + CHR$(34)) + ModifyCOMMAND$
-        IF path.exe$ = "" THEN path.exe$ = "./"
-        IF os$ = "LNX" THEN
+
+    SELECT CASE os$
+        CASE "WIN": SHELL _DONTWAIT QuotedFilename$(CHR$(34) + lastBinaryGenerated$ + CHR$(34)) + ModifyCOMMAND$
+
+        CASE "LNX"
+
+            IF path.exe$ = "" THEN path.exe$ = "./" './ to specify relative paths
+
             IF LEFT$(lastBinaryGenerated$, LEN(path.exe$)) = path.exe$ THEN
-                SHELL _DONTWAIT QuotedFilename$(lastBinaryGenerated$) + ModifyCOMMAND$
+                shellcmdline$ = QuotedFilename$(lastBinaryGenerated$) + ModifyCOMMAND$
             ELSE
-                SHELL _DONTWAIT QuotedFilename$(path.exe$ + lastBinaryGenerated$) + ModifyCOMMAND$
+                shellcmdline$ = QuotedFilename$(path.exe$ + lastBinaryGenerated$) + ModifyCOMMAND$
             END IF
-        END IF
-        IF path.exe$ = "./" THEN path.exe$ = ""
-    ELSE
-        IF os$ = "WIN" THEN SHELL QuotedFilename$(CHR$(34) + lastBinaryGenerated$ + CHR$(34)) + ModifyCOMMAND$
-        IF path.exe$ = "" THEN path.exe$ = "./"
-        IF os$ = "LNX" THEN
-            IF LEFT$(lastBinaryGenerated$, LEN(path.exe$)) = path.exe$ THEN
-                SHELL QuotedFilename$(lastBinaryGenerated$) + ModifyCOMMAND$
-            ELSE
-                SHELL QuotedFilename$(path.exe$ + lastBinaryGenerated$) + ModifyCOMMAND$
+
+            IF path.exe$ = "./" THEN path.exe$ = "" 'restore it to empty string
+
+
+            IF Console THEN
+                'launch terminal with program when using $CONSOLE
+                IF ReadConfigSetting(generalSettingsSection$, "DefaultTerminal", term$) = 0 THEN 'if DefaultTerminal is not in config.ini, then add it
+                    WriteConfigSetting generalSettingsSection$, "DefaultTerminal", "Auto"
+                    term$ = "auto"
+                END IF
+
+                IF LCASE$(term$) = "auto" THEN term$ = "$(./internal/support/auto-term.sh)"
+
+                shellcmdline$ = term$ + " -e " + shellcmdline$
+                PRINT shellcmdline$
+                SHELL _DONTWAIT shellcmdline$ 'bring up a terminal when using $Console
+
+                'not using $CONSOLE
+            ELSE SHELL _DONTWAIT shellcmdline$
             END IF
-        END IF
-        IF path.exe$ = "./" THEN path.exe$ = ""
-        DO: LOOP UNTIL INKEY$ = ""
-        DO: LOOP UNTIL _KEYHIT = 0

```

Comment:
Should these lines be replaced with _KEYCLEAR? As is it appears they've just been deleted.

Reply:
that was a part of an old debug thing that fellippe said it was safe to delete as those lines wouldn't be run anyway